include/dxc/DxilContainer/DxilPipelineStateValidation.h

```diff
@@ -937,12 +942,15 @@ DxilPipelineStateValidation::ReadOrWrite(const void *pBits, uint32_t *pSize,
 
   PSV_RETB(rw.MapValue(&m_uPSVRuntimeInfoSize, initInfo.RuntimeInfoSize()));
   PSV_RETB(rw.MapArray(&m_pPSVRuntimeInfo0, 1, m_uPSVRuntimeInfoSize));
-  AssignDerived(&m_pPSVRuntimeInfo1, m_pPSVRuntimeInfo0,
-                m_uPSVRuntimeInfoSize); // failure ok
-  AssignDerived(&m_pPSVRuntimeInfo2, m_pPSVRuntimeInfo0,
-                m_uPSVRuntimeInfoSize); // failure ok
-  AssignDerived(&m_pPSVRuntimeInfo3, m_pPSVRuntimeInfo0,
-                m_uPSVRuntimeInfoSize); // failure ok
+  if (initInfo.PSVVersion > 0)
+    AssignDerived(&m_pPSVRuntimeInfo1, m_pPSVRuntimeInfo0,
+                  m_uPSVRuntimeInfoSize); // failure ok
+  if (initInfo.PSVVersion > 1)
+    AssignDerived(&m_pPSVRuntimeInfo2, m_pPSVRuntimeInfo0,
+                  m_uPSVRuntimeInfoSize); // failure ok
+  if (initInfo.PSVVersion > 2)
```

Comment:
These should be unnecessary.  The scenarios are:
- CalcSize or Write mode: `m_uPSVRuntimeInfoSize` based on `PSVVersion` in `initInfo`, `AssignDerived` will only assign when struct fits in size.
- Read mode: `m_uPSVRuntimeInfoSize` determines the struct version available, PSVVersion in initInfo is not used for this, and there is no independent source for PSVVersion for PSV readers.

Validator will want to validate that `m_uPSVRuntimeInfoSize` matches the one computed by `initInfo.RuntimeInfoSize()` where `initInfo.PSVVersion` has been set according to validator version.  It's probably best if this is done manually by the validator before invoking `InitFromPSV0`, since an incorrect size will run that off the rails in a way that might just result in failure to load (malformed), or maybe not any error depending on the data and sizes, rather than a more specific error that we could emit in the validator.  If this size is wrong, there's no point in trying to read anything else.